### ISSUE TITLE
save_when feature

### ISSUE THREAD

**Issue by su-hy** [ansible-collections/cisco.asa#44]
##### SUMMARY
It would be nice if asa_config module had save_when functionality present in ios_config module. 
https://docs.ansible.com/ansible/latest/modules/ios_config_module.html#parameter-save_when

##### ISSUE TYPE
I would like to be able to save config only if it has been modified. 

##### COMPONENT NAME
asa_config

##### ADDITIONAL INFORMATION
It would work exactly like it works on ios_config, including the diff_ignore_lines

```yaml
tasks:
   - name: Save config to flash if it has been modified.
     ios_config:
      diff_against: startup
      diff_ignore_lines:
       - '[a-zA-Z0-9]{8}\s[a-zA-Z0-9]{8}\s.*'
       - 'certificate self-signed 01'
       - 'quit'
      save_when: modified
```

**Comment by venu1428** [comment]
hi @su-hy  is this feature got added to asa_config
I still did not see it in 2.10 version also

**Comment by justjais** [comment]
@su-hy @venu1428 I have raised the PR to add the asked feature from the `asa_config` module end, can you please verify if the PR changes work as you expect.

**Comment by su-hy** [comment]
@justjais I would test it, but I am a beginner in Ansible, I don't really know how to install that 1.1.0 version. From what I can tell I still have 1.0 on my system...

**Comment by justjais** [comment]
@su-hy `save_when` fix PR is yet not merged, so if feasible can you check the code changes from the PR and apply them locally and run the scenario which will help to verify if the fix is working as expected for you.

**Comment by robertegithub** [comment]
Has this been released? It's not documented with the 2.0 documentation for cisco.asa.asa_config linked from the main page (https://galaxy.ansible.com/cisco/asa)

**Comment by justjais** [comment]
@robertegithub I've raised the PR to add the particular feature, so if you can test out the PR #75 changes and update if that works for you as expected, then we can merge the same and release with this months asa collection release.

**Comment by su-hy** [comment]
@justjais I would test this for you but I am new to Ansible and I don't know which file I should download / replace to test the version you submitted. If you can provide some basic instructions I can try to test.

**Comment by robertegithub** [comment]
Thank you @justjais. Here is the test I performed:

1) Back up/move the original asa_config.py module in ~/.ansible/collections/ansible_collections/cisco/asa/plugins/modules/
Add the updated file from PR #75 . 

2) Run a test without changes to the Ansible playbook, still using "save: yes" and "save: no" in some locations.
Expected output: No difference
Result: Success

3) Leave the ASA config as is, so that no changes are required.
Modify the playbooks to use "save_when: changed" in places where "save: yes" is used.
Expected output: No unnecessary saves / no gratuitous 'wr mem' actions observed in "terminal monitor"
Result: Success

4) Update the ASA config so some changes will be required.
Run the playbook again with the "save_when: changed" parameter in place.
Expected output: Changes will be implemented and saved ('wr mem' actions will take place)
Result: Success

I did not test any of the other three arguments to save_when, but the "changed" argument is good.

**Comment by robertegithub** [comment]
Awesome. Thanks @justjais !!

**Comment by robertegithub** [comment]
Just bumping this - looks like it hasn't been released yet. Any updates on possible release for this feature?

**Comment by justjais** [comment]
FIx with the PR is merged and released with the ASA collection version `2.0.1`
